sinabs/backend/dynapcnn/nir_graph_extractor.py

```diff
@@ -0,0 +1,784 @@
+# author    : Willian Soares Girao
+# contact   : wsoaresgirao@gmail.com
+
+from copy import deepcopy
+from pprint import pformat
+from typing import Callable, Dict, Iterable, List, Optional, Set, Tuple, Type, Union
+
+import nirtorch
+import torch
+import torch.nn as nn
+
+from sinabs import layers as sl
+
+from .connectivity_specs import (
+    LAYER_TYPES_WITH_MULTIPLE_INPUTS,
+    LAYER_TYPES_WITH_MULTIPLE_OUTPUTS,
+    SupportedNodeTypes,
+)
+from .dvs_layer import DVSLayer
+from .dynapcnn_layer_utils import construct_dynapcnnlayers_from_mapper
+from .dynapcnnnetwork_module import DynapcnnNetworkModule
+from .exceptions import InvalidGraphStructure, UnsupportedLayerType
+from .sinabs_edges_handler import (
+    collect_dynapcnn_layer_info,
+    fix_dvs_module_edges,
+    handle_batchnorm_nodes,
+)
+from .utils import Edge, topological_sorting
+
+try:
+    from nirtorch.graph import TorchGraph
+except ImportError:
+    # In older nirtorch versions TorchGraph is called Graph
```

Comment:
The comment explains why there is this try/except statement. Without it it will break with older nirTorch versions. To be clear: This "older" version is still the default one that gets installed when running `pip install nirtorch`

Reply:
- Make sure nirtorch is in requirements.

Reply:
Hi @bauerfe, this class `nir_graph_extractor` is used only by this new implementation, or is it also used by older code?
I was about to create a new issue but found myself confused about the problem.

I don't remember if we said this is only necessary for this new implementation and that's why it shouldn't affect general requirement of sinabs, or what we conclude.